src/app/app.component.ts

```diff
@@ -198,51 +204,91 @@ export class AppComponent implements AfterViewInit {
-    });
+    let focusedNode;
+    if (this.lfSelectedFolder) {
+      const repoId = await this.repoClient.getCurrentRepoId();
+      const focusNodeByPath = await this.repoClient.entriesClient.getEntryByPath({
+          repoId: repoId,
+          fullPath: this.lfSelectedFolder.selectedFolderPath
+        });
+        const focusedNodeEntry = focusNodeByPath?.entry;
+        if (focusedNodeEntry) {
+          focusedNode = {
+            id: focusedNodeEntry.id.toString(),
+            isContainer: focusedNodeEntry.isContainer,
+            isLeaf: focusedNodeEntry.isLeaf,
+            path: this.lfSelectedFolder.selectedFolderPath,
+            name: focusedNodeEntry.name,
+          };
+        }
+    }
+    await this.lfRepositoryBrowser?.nativeElement.initAsync(this.lfRepoTreeNodeService, focusedNode as LfRepoTreeNode);
   }
 
+  isNodeSelectable = (node: LfRepoTreeNode) => {
+    if (node.entryType == EntryType.Folder) {
+      return true;
+    }
+    else {
+      return false;
+    }
+  };
+
   get isLoggedIn(): boolean {
     return this.loginComponent?.nativeElement?.state === LoginState.LoggedIn;
   }
 
   // Tree event handler methods
-  async onOkClick(okClickEvent: Event) {
-    const selectedNode = (okClickEvent as CustomEvent<TreeNode>).detail;
-    const breadcrumbs = this.lfFolderBrowserElement?.nativeElement?.breadcrumbs;
-    const entryId = Number.parseInt(selectedNode.id, 10);
-    const path = selectedNode.path;
-    this.selectedFolder = {
-      entryId,
-      path,
-      displayName: this.getFolderNameText(entryId, path),
-      displayPath: this.getFolderPathTooltip(path)
-    };
-    if (breadcrumbs) {
-      this.selectedFolder.breadcrumbs = breadcrumbs;
+  async onSelectFolder() {
+    const selectedNode = this.lfRepositoryBrowser.nativeElement.currentFolder as LfRepoTreeNode;
+    let entryId = Number.parseInt(selectedNode.id, 10);
+    const selectedFolderPath = selectedNode.path;
+    if (selectedNode.entryType == EntryType.Shortcut) {
+      entryId = selectedNode.targetId;
     }
-    const nodeId = selectedNode.id;
     const repoId = (await this.repoClient.getCurrentRepoId());
     const waUrl = this.loginComponent.nativeElement.account_endpoints.webClientUrl;
-    this.selectedNodeUrl = getEntryWebAccessUrl(nodeId, repoId, waUrl, selectedNode.isContainer);
     this.expandFolderBrowser = false;
+    this.lfSelectedFolder = {
+      selectedNodeUrl: getEntryWebAccessUrl(entryId.toString(), repoId, waUrl, selectedNode.isContainer),
+      selectedFolderName: this.getFolderNameText(entryId, selectedFolderPath),
+      selectedFolderPath: selectedFolderPath
+    }
+  }
+
+  get shouldShowSelect(): boolean {
+    return !this.shouldShowOpen && !!this.lfRepositoryBrowser?.nativeElement?.currentFolder;
+  }
+
+  get shouldShowOpen(): boolean {
+    return !!this.entrySelected;
+  }
+
+  onEntrySelected(event) {
+    const customEvent = event as CustomEvent<LfTreeNode[]>;
```

Comment:
do we really need to do anything onEntrySelected

Reply:
we can set this.entrySelected which is used to determine whether or not to show the open button

Reply:
maybe another way is to subscribe to the lfFieldContainerRef property, and once it becomes available, we can call initAsync?